tests/unit-tests/Queue/JobTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the hyn/multi-tenant package.
+ *
+ * (c) Daniël Klabbers <daniel@klabbers.email>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @see https://laravel-tenancy.com
+ * @see https://github.com/hyn/multi-tenant
+ */
+
+namespace Hyn\Tenancy\Tests\Queue;
+
+use Hyn\Tenancy\Contracts\CurrentHostname;
+use Hyn\Tenancy\Environment;
+use Hyn\Tenancy\Tests\Extend\JobExtend;
+use Hyn\Tenancy\Tests\Test;
+use Illuminate\Contracts\Foundation\Application;
+
+class JobTest extends Test
+{
+    /**
+     * @var Environment
+     */
+    protected $environment;
+
+    protected function duringSetUp(Application $app)
+    {
+        $this->setUpHostnames(true);
+        $this->setUpWebsites(true, true);
+        $this->environment = $app->make(Environment::class);
+    }
+
+    /**
+     * @test
+     */
+    public function serializes_tenant()
+    {
+        $this->app->make(CurrentHostname::class);
+
+        $job = new JobExtend();
+
+        $attributes = serialize($job);
+
+        // switch to other
+        $this->environment->hostname($this->tenant);
+
+        /** @var CurrentHostname $identified */
+        $identified = $this->app->make(CurrentHostname::class);
+
+        /** @var JobExtend $job */
+        $job = unserialize($attributes);
+
+        $this->assertInstanceOf(JobExtend::class, $job);
+
+        $restored = $this->environment->hostname();
+
+        $this->assertNotEquals($identified->fqdn, $restored->fqdn);
+    }
```

Comment:
I'd like to see a test that injects some things in the database. Just to be sure. I like how you created this test, but I would be much comfortable if the data injected is also injected to the right database.

Reply:
I understand I would have loved to do the same. But.. to do a full fledged, real life test would require the Job process to be executed outside of the current process (to ensure no other tenant was previously activated). I could come up with a solution but the current test already proves the implementation works as is.